node/src/components/consensus/highway_core/synchronizer.rs

```diff
@@ -409,8 +426,10 @@ impl<I: NodeIdT, C: Context + 'static> Synchronizer<I, C> {
                     continue;
                 }
                 // Otherwise request the missing dependency from the sender.
-                debug!(dependency = ?transitive_dependency, %sender, "requesting dependency");
-                let ser_msg = HighwayMessage::RequestDependency(transitive_dependency).serialize();
+                let uuid = thread_rng().next_u64();
```

Comment:
Using `thread_rng()` is undesirable, especially in production code.  Components get passed an RNG via `handle_event()` to avoid using a variety of different RNGs.  It also means in tests we generally get reproducibility.

Same comment applies to a couple of places in highway.rs too.

Reply:
I did consider that but didn't see any real advantage of doing that. It was adding quite a lot of complexity to the code (as none of the highway/consensus top-level methods are accepting it now) and wasn't useful anywhere else (we don't need that kind of determinism in the test code).